api/src/test/java/org/openmrs/module/appointments/service/AppointmentsServiceTest.java

```diff
@@ -67,6 +72,24 @@ public void init() throws Exception {
         executeDataSet("appointmentTestData.xml");
     }
 
+    @Test
+    public void shouldSaveAppointmentsOnlyIfUserHasManageOwnPrivilege() throws Exception {
+        Context.authenticate(manageOwnUser, manageOwnUserPassword);
+        Appointment appointment = getSampleAppointment();
+        assertNotNull(appointmentsService.validateAndSave(appointment));
+    }
+
```

Comment:
Can you add a test that shows that if a provider is editing an appointment they are not a provider for it will fail?

Reply:
Sure . Will add those tests

Reply:
The tests are already there for these scenarios in AppointmentsServiceImplTest class. Is there any specific reason we need to mention the tests at this level.

Reply:
no you don't have to mention them twice. I probably missed them in my initial review. Will look again.